File: gto-support-jsonapi/src/main/java/org/ccci/gto/android/common/jsonapi/model/JsonApiError.java
package org.ccci.gto.android.common.jsonapi.model;

import androidx.annotation.Nullable;

import org.json.JSONObject;

public class JsonApiError {
    public static final String JSON_ERROR_STATUS = "status";
    public static final String JSON_ERROR_DETAIL = "detail";
    public static final String JSON_ERROR_SOURCE = "source";
    public static final String JSON_ERROR_META = "meta";

    @Nullable
    private Integer mStatus;
    @Nullable
    private String mDetail;

    @Nullable
    private Source mSource;

    @Nullable
    private JSONObject mRawMeta;

    @Nullable
    public String getDetail() {
        return mDetail;
    }

    public void setDetail(@Nullable final String detail) {
        mDetail = detail;
    }

    @Nullable
    public Integer getStatus() {
        return mStatus;
    }

    public void setStatus(@Nullable final Integer status) {
        mStatus = status;
    }

    @Nullable
    public Source getSource() {
        return mSource;
    }

    public void setSource(@Nullable final Source source) {
        mSource = source;
    }

    @Nullable
    public JSONObject getRawMeta() {
        return mRawMeta;
    }

    public void setRawMeta(@Nullable final JSONObject meta) {
        mRawMeta = meta;
    }

    public static class Source {
        public static final String JSON_ERROR_SOURCE_POINTER = "pointer";

        @Nullable
        private String mPointer;

        @Nullable
        public String getPointer() {
            return mPointer;
        }

        public void setPointer(@Nullable final String pointer) {
            mPointer = pointer;
        }
    }
}

File: gto-support-jsonapi/src/main/java/org/ccci/gto/android/common/jsonapi/model/JsonApiError.kt
package org.ccci.gto.android.common.jsonapi.model

import org.json.JSONObject

class JsonApiError {
    companion object {
        const val JSON_ERROR_STATUS = "status"
        const val JSON_ERROR_TITLE = "title"
        const val JSON_ERROR_DETAIL = "detail"
        const val JSON_ERROR_SOURCE = "source"
        const val JSON_ERROR_META = "meta"
    }

    var status: Int? = null
    var title: String? = null
    var detail: String? = null
    var source: Source? = null
    var rawMeta: JSONObject? = null

    class Source {
        companion object {
            const val JSON_ERROR_SOURCE_POINTER = "pointer"
        }

        var pointer: String? = null
    }
}
